src/main/proto/catalogue.proto

```diff
@@ -0,0 +1,244 @@
+syntax="proto3";
+
+option java_multiple_files = true;
+option java_package = "org.arjun.CatalogueMicroservice";
+
+import "google/protobuf/empty.proto";
+
+//Services related to Item
+service ItemService {
+  //Add an item to the DB
+  rpc CreateItem(CreateItemRequest) returns (Item);
+
+  //Add a stream of Items to the DB
+  rpc CreateItemsStream(stream CreateItemsStreamRequest) returns (ItemsStream);
+
+  //Delete a specific Item
+  rpc DeleteItem(DeleteItemRequest) returns (google.protobuf.Empty);
+
+  //Get an specific Item
+  rpc GetItem(GetItemRequest) returns (Item);
+  //Get all items
+  rpc GetItems(GetItemsRequest) returns (stream Items);
+  //Get all items of a specific catalogue
+  rpc GetItemsCatalogue(GetItemsCatalogueRequest) returns (stream ItemsCatalogue);
+  //Get all items of a specific user
+  rpc GetItemsUser(GetItemsUserRequest) returns (stream ItemsUser);
```

Comment:
use a oneof in `GetItemsRequest` for the other 2 rpcs as well

Reply:
Used oneof feature